couchbase/src/main/java/org/hibernate/ogm/datastore/couchbase/util/impl/DatabaseIdentifier.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Hibernate OGM, Domain model persistence for NoSQL datastores
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.ogm.datastore.couchbase.util.impl;
+import java.net.MalformedURLException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.net.URL;
+public class DatabaseIdentifier {
+	private static final String PROTOCOL = "http";
+	private static final String SLASH = "/";
+	private final String host;
+	private final int port;
+	private final String databaseName;
+	private final String password;
+	private final URI serverUri;
+	private final URI databaseUri;
+	public DatabaseIdentifier(String host, int port, String databaseName, String password)
```

Comment:
Seems like its copied from the CouchDB backend? It'd be nice if we could unify it.